permhash/helpers.py

```diff
@@ -199,18 +219,29 @@ def create_crx_manifest_permlist(path):
     Creates and returns the list of permissions that will be used to create the permhash.
 
     :param path: The path to the file where the permissions need to be retrieved
-    :type perm_list: string
+    :type path: string
     """
     # Ensure the file exists, is non-zero in size, and is a mimetype that a CRX should be
     if check_type(path, CRX_MANIFEST_MIMETYPES):
-        with open(path, encoding="utf-8") as manifest:
-            try:
-                manifest_json = json.load(manifest)
-            except OSError:
-                logging.warning("Failure to load JSON from the manifest at %s", path)
-                return False
-            perm_list = parse_crx_manifest(manifest_json)
-            return perm_list
+        with open(path, "rb") as manifest_byte_stream:
+            if manifest_byte_stream.readable():
+                try:
+                    manifest_byte_read = manifest_byte_stream.read()
+                except IOError as error:
+                    # This file was unable to be read
+                    logging.warning("This manifest is unable to be read.")
+                    logging.warning(str(error))
+                    return False
+                processed_json = process_manifest_bytes(manifest_byte_read)
+                if not processed_json:
+                    logging.warning("Path: %s", path)
```

Comment:
Should probably return here right? Otherwise we'll pass the value false into `parse_crx_manifest`